data/metric/metrics.go

```diff
@@ -14,6 +15,12 @@ import (
 // each one.
 type SourceType int
 
+// Attribute represents an attribute metric in key-value pair format.
+type Attribute struct {
```

Comment:
Why not declaring Attribute as `map[string]string` ? Or why not using directly a map as users will not need to use that custom type?

Reply:
Oh I see. You want to keep the previous interface with no attributes specified.

Reply:
If we don't enforce users to set those values, they will have eventually the same issue we are trying to fix here.

Reply:
The point on types (besides safety) is expressing intention. So you understand what that data is and what is aimed to do.

Reply:
The point on enforcing that was discussed with @areina.

We agreed that there is no need for that as most of the users won't need the feature, so enforcing them to solve a problem they don't have is adding undesired complexity.

Reply:
IMHO we should enforce it since the user is not aware of the real problem when using RATE and DELTA: they may be facing the issue without even being aware of it.

Reply:
The big issue here, and what I consider we must fix asap is How to friendly let the user know that the metric sets should be unique.

Reply:
Added error as a warning (does not breaks the logic) when SetMetric on rate/gauge in a metric-set without attributes

Reply:
https://github.com/newrelic/infra-integrations-sdk/pull/106/files#diff-83416e4d39b0688a4bec5f4a2c076486R98